src/ota_image_libs/v1/file_table/db.py

```diff
@@ -290,13 +297,11 @@ def select_all_digests_with_size(
                 _stmt=stmt,
             )
 
-    def iter_dir_entries(self) -> Generator[DirTypedDict]:
+    def iter_dir_entries(self) -> Generator[DirRow]:
         with FileTableDirORM(self.connect_fstable_db()) as orm:
-            _row_factory = typing.cast(Callable[..., DirTypedDict], sqlite3.Row)
-
             # fmt: off
             yield from orm.orm_select_entries(
-                _row_factory=_row_factory,
+                _row_factory=DirRow.table_row_factory2,
```

Comment:
(nits) the `table_row_factory2` will not be good name. will be better to change the name in the future(e.g `table_row_factory_strict` and `table_row_factory_lenient`)